test/unit/core/DatetimeClasses.js

```diff
@@ -110,21 +111,43 @@ describe('DateTimeClassesTest', function () {
             (() => LocalTime.fromString(null)).should.throw(TypeError, 'String expected');
             (() => LocalTime.fromString()).should.throw(TypeError, 'String expected');
         });
+
+        it('should construct from fromDate correctly', function () {
+            const localTime1 = LocalTime.fromDate(new Date(2000, 2, 29, 0, 0, 0, 0));
+            localTime1.toString().should.be.eq('00:00:00');
+            const localTime2 = LocalTime.fromDate(new Date(2000, 0, 29, 2, 3, 4, 6));
+            localTime2.toString().should.be.eq('02:03:04.006000000');
+        });
+
+        it('should throw when constructed from fromDate with a non-date thing', function () {
+            const nonDateThings = [1, null, '', {}, [], function() {}, class A {}, LocalDateTime.fromDate(new Date())];
+            nonDateThings.forEach(nonDateThing => {
+                (() => LocalTime.fromDate(nonDateThing)).should.throw(TypeError, 'A Date is not passed');
+            });
+        });
+
+        it('should throw when constructed from fromDate with an invalid date', function () {
+            const invalidDates = [new Date('aa'), new Date({}), new Date(undefined)];
+            invalidDates.forEach(invalidDate => {
+                isNaN(invalidDate.getTime).should.be.true;
+                (() => LocalTime.fromDate(invalidDate)).should.throw(RangeError, 'Invalid Date is passed');
+            });
+        });
     });
     describe('LocalDateTest', function () {
-        it('should throw RangeError if year is not an integer between -999_999_999-999_999_999(inclusive)',
-            function () {
-                (() => new LocalDate(1e9, 1, 1)).should.throw(RangeError, 'Year');
-                (() => new LocalDate(-1e9, 1, 1)).should.throw(RangeError, 'Year');
-                (() => new LocalDate(1.1, 1, 1)).should.throw(RangeError, 'All arguments must be integers');
-                (() => new LocalDate('1', 1, 1)).should.throw(TypeError, 'All arguments must be numbers');
-                (() => new LocalDate({ 1: 1 }, 1, 1)).should.throw(TypeError, 'All arguments must be numbers');
-                (() => new LocalDate([], 1, 1)).should.throw(TypeError, 'All arguments must be numbers');
-                (() => new LocalDate(1e12, 1, 1)).should.throw(RangeError, 'Year');
-            });
+        it('should throw RangeError if year is not an integer between -999_999_999-999_999_999(inclusive)', function () {
```

Comment:
should we also add boundary test, e.g. 
`new LocalDate(-999_999_999, 1, 1)` should not throw. 
Similar for other tests as well. I assume the these ranges are same as Java ranges, right?

Reply:
>  I assume the these ranges are same as Java ranges, right?
Yes. 


>should we also add boundary test, e.g.
new LocalDate(-999_999_999, 1, 1) should not throw.

We did not test all boundary values but I see a boundary test on line 173. We can add them later.